Sources/Sentry/SentryAppState.m

```diff
@@ -1,16 +1,26 @@
 #import <Foundation/Foundation.h>
+#import <NSDate+SentryExtras.h>
 #import <SentryAppState.h>
 
+NS_ASSUME_NONNULL_BEGIN
+
 @implementation SentryAppState
 
 - (instancetype)initWithReleaseName:(NSString *)releaseName
                           osVersion:(NSString *)osVersion
                         isDebugging:(BOOL)isDebugging
+                systemBootTimestamp:(NSDate *)systemBootTimestamp
 {
     if (self = [super init]) {
         _releaseName = releaseName;
         _osVersion = osVersion;
         _isDebugging = isDebugging;
+
+        // Round down to seconds as the precision of the serialization of the date is only
+        // milliseconds. With this we avoid getting different dates before and after serialization.
```

Comment:
not sure if I get it, does it mean we're losing the millis precision? if so, I'd not go this way, metrics should have high precision

Reply:
We don't use the `systemBootTimestamp` for the calculation of the measurement. We only use it to identify when the system was rebooted. This is fine for now, I would say.